docs/tutorials/getting-started-with-node-js.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Getting Started with Node JS
+group: software-languages
+author: "@liam-b"
+---
+
+{% include /style/icon.html type="warning" %}
+Note that Node.js runs frustratingly slowly on the EV3 brick and is limited to an old version of both Node.js and npm. You may want to consider using Python or another ev3dev-supported language instead.
+{: .alert .alert-warning}
+
+<p class="lead">
+Node.js is a framework that allows you to run JavaScript as a local application. It is designed to make building servers and other asynchronous apps as easy as possible. In this tutorial, we'll show you how to get up-and-running with Node.js on ev3dev.
+</p>
+
+**Before you start, make sure that you have [configured a network connection](/docs/networking/) on
+your ev3dev device and have [opened an SSH connection](/docs/tutorials/connecting-to-ev3dev-with-ssh/) to it.**
+
+## Installing ev3dev-lang (and quick-start)
+To install the library which allows you to use ev3dev features, run the following command from your SSH connection:
+
+```shell
+npm install ev3dev-lang
+```
+
+In all JavaScript files that you write from here forward, include the following line to import the module:
+```javascript
+var ev3dev = require('ev3dev-lang');
+```
+
+For a bit more info on this visit the ev3dev-lang-js [GitHub page](https://github.com/wasabifan/ev3dev-lang-js).
+
+## Exploring Node.js and ev3dev
+
+### Picking an editor: Nano
+
+For this tutorial, we will be using a text editor called `nano`. For more info on how to use nano, check out the [nano cheat sheet](/docs/tutorials/nano-cheat-sheet). Feel free to use a different one if you know what you're doing.
+
+### Hello world
+To start, navigate to a project directory (e.g `~/src/js`) that we can store our files in:
+
+```shell
+mkdir -p ~/src/js && cd ~/src/js
+```
+
+Create a JavaScript file and open it with `nano test.js`. In the opened editor, add the following code:
+
+```javascript
+console.log('hello world!');
+```
+
+If you're using `nano`, refer to the [nano cheat sheet](/docs/tutorials/nano-cheat-sheet/) to learn how to save the file.
+
+Running `node test.js`should print `hello world!` to the console. If this worked, you are ready to start coding with node!
+
+### Motors
+Let's try running a motor. Plug an EV3 motor into port A, then open `test.js` again and write:
+
+```javascript
+var ev3dev = require('ev3dev-lang');
+
+var motor = new ev3dev.Motor('outA'); // create new motor plugged into port A called 'motor'
```

Comment:
Might want a link to <http://www.ev3dev.org/docs/platform-comparison/> here to get the port names for non-EV3 hardware.